rewrite-javascript/src/main/java/org/openrewrite/javascript/tree/JS.java

```diff
@@ -4548,4 +4546,163 @@ public ExportSpecifier withTypeOnly(JLeftPadded<Boolean> typeOnly) {
         }
 
     }
+    @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+    @EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
+    @RequiredArgsConstructor
+    @AllArgsConstructor(access = AccessLevel.PRIVATE)
+    final class IndexedAccessType implements JS, Expression, TypeTree {
+        @Nullable
+        @NonFinal
+        transient WeakReference<IndexedAccessType.Padding> padding;
+
+        @With
+        @EqualsAndHashCode.Include
+        @Getter
+        UUID id;
+
+        @With
+        @Getter
+        Space prefix;
+
+        @With
+        @Getter
+        Markers markers;
+
+        /**
+         * (prefix)objectType(rightPaddedSuffix)[(prefix)indexType(suffix)](rightPaddedSuffix)
```

Comment:
@OlegDokuka This is the case we discussed yesterday. I am not quite sure what the `(suffix)` is, but assuming it would be the `JRightPadded#after` of the `indexType` I think that would be correct. Now the only thing I see missing here then is the `JRightPadded` to wrap the `objectType` property.